Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixConstraint.h

```diff
@@ -0,0 +1,701 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/linearalgebra/config.h>
+#include <sofa/linearalgebra/CompressedRowSparseMatrixGeneric.h>
+
+
+namespace sofa::linearalgebra
+{
+
+template<class RowType, class VecDeriv, typename Real = typename VecDeriv::value_type::Real>
+Real CompressedRowSparseMatrixVecDerivMult(const RowType row, const VecDeriv& vec)
+{
+    Real r = 0;
+    for (typename RowType::first_type it = row.first, itend = row.second; it != itend; ++it)
+        r += it.val() * vec[it.index()];
```

Comment:
`std::accumulate` can be used

Reply:
I dont think it is possible, `it` is a `ColConstIterator` (which iterates on Index) and the loop needs the iterator itself to get val()